core/src/main/java/org/elasticsearch/search/fetch/subphase/InnerHitsContext.java

```diff
@@ -112,20 +136,29 @@ public NestedInnerHits(String name, SearchContext context, ObjectMapper parentOb
         }
 
         @Override
-        public TopDocs topDocs(SearchContext context, FetchSubPhase.HitContext hitContext) throws IOException {
+        public TopDocs topDocs(SearchContext context, SearchHit hit) throws IOException {
```

Comment:
or alternatively would it work to remove the search context from this signature and use the one from the constructor to create the weight?